tests/e2e/multi_vc.go

```diff
@@ -0,0 +1,1827 @@
+		scParameters                map[string]string
+		topValStartIndex            int
+		topValEndIndex              int
+		topkeyStartIndex            int
+		datastoreURLVC1             string
+		datastoreURLVC2             string
+		podAntiAffinityToSet        bool
+		sshClientConfig             *ssh.ClientConfig
+		nimbusGeneratedK8sVmPwd     string
+		allMasterIps                []string
+		masterIp                    string
+		scaleUpReplicaCount         int32
+		scaleDownReplicaCount       int32
+		multiVCSetupType            string
+		isVsanHealthServiceStopped  bool
+		stsScaleUp                  bool
+		stsScaleDown                bool
+		verifyTopologyAffinity      bool
+		storagePolicyInVc1          string
+		storagePolicyInVc1Vc2       string
+		storagePolicyToDelete       string
+		isSPSServiceStopped         bool
+	)
+	ginkgo.BeforeEach(func() {
+		var cancel context.CancelFunc
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+
+		multiVCbootstrap()
+
+		stsScaleUp = true
+		stsScaleDown = true
+		verifyTopologyAffinity = true
+
+		sc, err := client.StorageV1().StorageClasses().Get(ctx, defaultNginxStorageClassName, metav1.GetOptions{})
+		if err == nil && sc != nil {
+			gomega.Expect(client.StorageV1().StorageClasses().Delete(ctx, sc.Name,
+				*metav1.NewDeleteOptions(0))).NotTo(gomega.HaveOccurred())
+		}
+
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+		topologyMap := GetAndExpectStringEnvVar(topologyMap)
+		allowedTopologies = createAllowedTopolgies(topologyMap, topologyLength)
+		bindingMode = storagev1.VolumeBindingWaitForFirstConsumer
+		scParameters = make(map[string]string)
+		storagePolicyInVc1 = GetAndExpectStringEnvVar(envStoragePolicyNameVC1)
+		storagePolicyInVc1Vc2 = GetAndExpectStringEnvVar(envStoragePolicyNameInVC1VC2)
+		storagePolicyToDelete = GetAndExpectStringEnvVar(envStoragePolicyNameToDeleteLater)
+		datastoreURLVC1 = GetAndExpectStringEnvVar(envSharedDatastoreURLVC1)
+		datastoreURLVC2 = GetAndExpectStringEnvVar(envSharedDatastoreURLVC2)
+		nimbusGeneratedK8sVmPwd = GetAndExpectStringEnvVar(nimbusK8sVmPwd)
+		multiVCSetupType = GetAndExpectStringEnvVar(envMultiVCSetupType)
+
+		sshClientConfig = &ssh.ClientConfig{
+			User: "root",
+			Auth: []ssh.AuthMethod{
+				ssh.Password(nimbusGeneratedK8sVmPwd),
+			},
+			HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		}
+
+		// fetching k8s master ip
+		allMasterIps = getK8sMasterIPs(ctx, client)
+		masterIp = allMasterIps[0]
+
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+		fss.DeleteAllStatefulSets(client, namespace)
```

Comment:
This util DeleteAllStatefulSets works perfectly fine when we have only a single statefulset in a given namespace.

I can add it, but it will not be required here because I am creating only 1 single StatefulSet in a given namespace..
This util deleteAllStatefulSetAndPVs is useful when we are creating multiple Statefulsets, Standalone pods, Deployments pods in a given namespace